sqlglot/expressions.py

```diff
@@ -3009,7 +3013,7 @@ def replace_children(expression, fun):
             else:
                 new_child_nodes.append(cn)
 
-        expression.args[k] = new_child_nodes if is_list_arg else new_child_nodes[0]
+        expression.args[k] = new_child_nodes if is_list_arg else list_get(new_child_nodes, 0)
```

Comment:
We can't leave this as `new_child_nodes[0]`: if the transformation returns `None`, we'll get an index error because the innermost loop won't be executed and `new_child_nodes` will remain an empty list.